src/FixtureBuilder/ExpressionLanguage/Lexer/ReferenceEscaperLexer.php

```diff
@@ -44,8 +44,7 @@ public function __construct(LexerInterface $decoratedLexer)
      */
     public function lex(string $value): array
     {
-        $escapedValue = preg_replace('/(\\p{L})@/', '$1\\@', $value);
-
+        $escapedValue = preg_replace('/((\s|^)[a-zA-Z0-9_.+-]+?[A-Za-z0-9])@/', '$1\\@', $value);
```

Comment:
@theofidry something like 
`((\s|^)[^@]?(\p{L}|\p{N}|[+-_.])+?(\p{L}|\p{N}))@` should work just fine.
I will test this later.